File: 3105/longest-strictly-increasing-or-strictly-decreasing-subarray.c
// LeetCode 3105. Longest Strictly Increasing or Strictly Decreasing Subarray
// 持續增加 or 持續減少，最長的長度是多少？
int longestMonotonicSubarray(int* nums, int numsSize) {
    int ans = 1, inc = 1, dec = 1;
    for(int i=1; i<numsSize; i++) {
        if(nums[i-1]<nums[i]) { // 持續增加
            inc++;
            if(inc>ans) ans = inc;
        } else inc = 1; // 沒有持續增加，重來

        if(nums[i-1]>nums[i]) { // 持續減少
            dec++;
            if(dec>ans) ans = dec;
        } else dec = 1; // 沒有持續減少，重來
    }
    return ans;
}

File: 3105/longest-strictly-increasing-or-strictly-decreasing-subarray.py
# LeetCode 3105. Longest Strictly Increasing or Strictly Decreasing Subarray
# 持續增加 or 持續減少，最長的長度是多少？
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        ans = inc = dec = 1
        for i in range(1,len(nums)):
            if nums[i-1] < nums[i]: inc += 1  # 增加
            else: inc = 1  # 沒有增加

            if nums[i-1] > nums[i]: dec += 1  # 減少
            else: dec = 1  # 沒有減少
            
            ans = max(ans, inc, dec)  # 最長的長度
        return ans
